bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -430,80 +372,171 @@ impl ClientBuilder {
         builder.system_is_memory_constrained = true;
         Arc::new(builder)
     }
+}
 
-    pub fn username(self: Arc<Self>, username: String) -> Arc<Self> {
+#[derive(Clone, uniffi::Object)]
+pub struct IndexedDbSessionBuilder {
+    #[cfg_attr(not(target_family = "wasm"), allow(unused))]
+    name: String,
+    passphrase: Option<String>,
+}
+
+#[matrix_sdk_ffi_macros::export]
+impl IndexedDbSessionBuilder {
+    #[uniffi::constructor]
+    pub fn new(name: String) -> Arc<Self> {
+        Arc::new(Self { name, passphrase: None })
+    }
+
+    /// Set the passphrase for the IndexedDB store.
+    pub fn passphrase(self: Arc<Self>, passphrase: Option<String>) -> Arc<Self> {
         let mut builder = unwrap_or_clone_arc(self);
-        builder.username = Some(username);
+        builder.passphrase = passphrase;
         Arc::new(builder)
     }
+}
 
-    pub fn server_name(self: Arc<Self>, server_name: String) -> Arc<Self> {
+#[derive(Clone)]
+pub enum ClientSessionConfig {
+    /// Setup the client to use the SQLite store.
+    #[cfg_attr(target_family = "wasm", allow(unused))]
+    Sqlite(SqliteSessionBuilder),
+
+    /// Setup the client to use the IndexedDB store.
+    #[cfg_attr(not(target_family = "wasm"), allow(unused))]
+    IndexedDb(IndexedDbSessionBuilder),
+}
```

Comment:
I think you can remove all this, and add:

```suggestion
#[cfg(target_family = "wasm")]
pub type SessionConfig = IndexedDbSessionBuilder;

#[cfg(not(target_family = "wasm"))]
pub type SessionConfig = SqliteSessionBuilder;
```